.github/workflows/fast-forward.yml

```diff
@@ -9,19 +9,60 @@ jobs:
     name: Fast Forward
     if: |
       (github.event.issue.pull_request != '') &&
-      github.event.comment.body == '/fast-forward' &&
-      (github.event.comment.author_association == 'MEMBER' || github.event.comment.author_association == 'OWNER')
+      github.event.comment.body == '/fast-forward'
     runs-on: ubuntu-latest
     steps:
+      - uses: actions-cool/check-user-permission@v2
+        id: has_permission
+        with:
+          require: 'write'
+
+      - name: Comment PR
+        uses: thollander/actions-comment-pull-request@v1
+        if: steps.has_permission.outputs.require-result != 'true'
+        with:
+          pr_number: ${{ github.event.issue.number }}
+          message: |
+            You have no write permission on this repository
+
+      - name: Exit if no write permission
+        if: steps.has_permission.outputs.require-result != 'true'
+        run: exit 1
+
+      - uses: actions/github-script@v6
+        id: current_pr
+        with:
+          script: |
+            return github.rest.pulls.get({
+              pull_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+            })
+
+      - run: echo $prJSON
+        env:
+          prJSON: ${{ toJSON(fromJSON(steps.current_pr.outputs.result).data) }}
+
+      - name: Comment PR
+        uses: thollander/actions-comment-pull-request@v1
+        if: fromJSON(steps.current_pr.outputs.result).data.mergeable_state != 'clean'
```

Comment:
Did you find any documentation for mergeable_state?

I only found this:
https://github.com/orgs/community/discussions/24299
https://docs.github.com/en/rest/pulls/pulls#get-a-pull-request
https://docs.github.com/en/graphql/reference/enums#mergestatestatus

The last one is from the GraphQL API and has a different property name, but values seem to match mergeable_state form the REST API. So the value is only 'clean' if the PR has no conflicts and all required status checks have passed, right?

Reply:
I did not test the case of a conflict, but then the fast forward would anyway fail.
What i tested is: the state is clean if all checks pass and there are enough approving reviews from people with write access.